### PR TITLE
Move uf model code from uf to quantifiers

### LINKED ISSUE #1302: Move classes from uf/theory_uf_model to quantifiers

### PR DESCRIPTION
**Pull request by ajreynol**
Fixes #1302.

### CODE REVIEW

File: src/theory/quantifiers/fmf/model_builder.cpp [hunk 1]
```diff
@@ -166,9 +166,81 @@ bool TermArgBasisTrie::addTerm(FirstOrderModel* fm, Node n, unsigned argIndex)
...
+Node QModelBuilderIG::UfModelPreferenceData::getBestDefaultValue(
+    Node defaultTerm, TheoryModel* m)
+{
+  Node defaultVal;
+  double maxScore = -1;
+  for (size_t i = 0, size = d_values.size(); i < size; i++)
+  {
+    Node v = d_values[i];
+    double score = (1.0 + (double)d_value_pro_con[0][v].size())
+                   / (1.0 + (double)d_value_pro_con[1][v].size());
```
Comment (HanielB) [inline src/theory/quantifiers/fmf/model_builder.cpp]: static_cast

File: src/theory/quantifiers/fmf/model_builder.cpp [hunk 2]
```diff
@@ -166,9 +166,81 @@ bool TermArgBasisTrie::addTerm(FirstOrderModel* fm, Node n, unsigned argIndex)
...
+      defaultVal = newDefaultVal;
+      Debug("fmf-model-cons-debug") << "-> Change default value to ";
+      m->printRepresentativeDebug("fmf-model-cons-debug", defaultVal);
+      Debug("fmf-model-cons-debug") << std::endl;
+    }
+    else
+    {
+      Debug("fmf-model-cons-debug")
+          << "-> Could not find arbitrary element of type "
+          << tn[(int)tn.getNumChildren() - 1] << std::endl;
```
Comment (HanielB) [inline src/theory/quantifiers/fmf/model_builder.cpp]: ditto

File: src/theory/quantifiers/first_order_model.cpp [hunk 3]
```diff
@@ -426,6 +426,121 @@ unsigned FirstOrderModel::getModelBasisArg(Node n)
...
+
+void FirstOrderModelIG::UfModelTreeGenerator::setValue(
+    TheoryModel* m, Node n, Node v, bool ground, bool isReq)
+{
+  Assert(!n.isNull());
+  Assert(!v.isNull());
+  d_set_values[isReq ? 1 : 0][ground ? 1 : 0][n] = v;
+  if (!ground)
+  {
+    unsigned defSize = d_defaults.size();
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply src/theory/quantifiers/first_order_model.cpp]: Where? defSize is used in the for directly below this line.

[the remark this answers is not in the document]
Comment (HanielB) [reply src/theory/quantifiers/first_order_model.cpp]: declare it in the for, I mean

**Pull request merged**